### ISSUE TITLE
Data shape and size

### ISSUE THREAD

**Issue by samanemami** [lyp22/ResSCNN#4]
May I kindly ask about the input size of the model?
I assume you are using a 3-Dimensional tensor.

I am unsure if the dimension of the `MinkowskiConvolution` layer is 2D or 1D.

Would you please help me with this?

Thank you

**Comment by lyp22** [comment]
The input is the so-called Minkowski tensor, which contains 2 tensors: one is a coordinate tenor [batch_index, X, Y, Z] and the other is a feature tensor [R, G, B].
MinkowskiConvolution is 3D, which is conducted on 3D coordinates.

**Comment by samanemami** [comment]
Thank you for the clear explanation.

May I ask what lib you used to transform a point cloud to `[batch_index, X, Y, Z] `? and how you defined the `batch_index`?

If you are feeding all the point clouds at the same time, we could have the MOS you already uploaded, but if we are learning one point cloud by one, the MOS should be a vector!

Thanks

**Comment by lyp22** [comment]
In this code, batch_index is always 0, because batch_size is set to 1. The reason is that for quality assessment, the down-sampling should be avoided, and thus the data is so big to process several point clouds at the same time.

In other applications, the batch_index is automatically generated in the `collate_fn` method in `torch.utils.data.DataLoader`. If defining `collate_fn` manually, `MinkowskiEngine.utils.batched_coordinates` can automatically assign batch_index for a coordinate tensor list.

For the second question, in this code, it is not "learning one point cloud by one". Instead, the point clouds are input into the GPU one by one. After inputting several samples and obtaining their predicted quality scores, the network is optimized at once. It is a process of "iterative optimization" instead of the conventional "batch optimization".

**Comment by samanemami** [comment]
> For the second question, in this code, it is not "learning one point cloud by one". Instead, the point clouds are input into the GPU one by one. After inputting several samples and obtaining their predicted quality scores, the network is optimized at once. It is a process of "iterative optimization" instead of the conventional "batch optimization".

Thank you so much.

But by point clouds, you are referring to different distortions of one point cloud?

**Comment by lyp22** [comment]
For example,
For a epoch:  [1.ply 2.ply 3.ply ...... 20000.ply]:
For 20000/8 iterations (a iteration which cotains 8 plys):
>1.ply  -> score 1
>2.ply  -> score 2
>......
>8.ply  -> score 8

The above process is serial input instead of parallel input.
Then conduct the optimization.
loss([score1, score2, ... , score8], [label1, label2, ... , label8]).backward()
END
END

**Comment by samanemami** [comment]
Thanks 
Therefore, with this assumption, if I add more rows with the name and path of the point cloud and MOS to Excel, is the model train based on all the images?

Now it only takes the `4armsMonstre_ColorNoise_2.ply` point cloud

**Comment by samanemami** [comment]
The first tensor of the list (from the data loader) is labeled as the feature. The second one is the coordinates, followed by the label MOS and the name of the file at the end.

Regarding the first dim of the above tensor which is the feature, how did you extract the features from the file?
I could not find it in the `ResSCNNDataset()`

**Comment by lyp22** [comment]
Yes

> Thanks Therefore, with this assumption, if I add more rows with the name and path of the point cloud and MOS to Excel, is the model train based on all the images?
> 
> Now it only takes the `4armsMonstre_ColorNoise_2.ply` point cloud

**Comment by lyp22** [comment]
It in `__getitem__` .  First, the point cloud filename is obtained. Then `open3d` is used to obtain the RGB and coordinate of the point cloud. Finally, they are converted into the tensor with the format required by MinkowskiTensor. And the MinkowskiTensor is generated during training and testing in trainer.py.

> The first tensor of the list (from the data loader) is labeled as the feature. The second one is the coordinates, followed by the label MOS and the name of the file at the end.
> 
> Regarding the first dim of the above tensor which is the feature, how did you extract the features from the file? I could not find it in the `ResSCNNDataset()`

**Comment by samanemami** [comment]
Regarding the data input, we only can enter one image with different distortions? Is it possible to have all the images and distortion at the same time?

One other question;
The result returns PLCC and SROCC for 100 optimizers epochs,
This is the epoch to fit the resNet?

Thank you

**Comment by lyp22** [comment]
If your GPU is better, you can enter several samples at the same time by modifying 'batch_size'.

Basically, 100 epochs can have a good result. You can also set 100 epochs to 1000 epochs and add a 'early stop' to control the termination of the training.

**Comment by samanemami** [comment]
Thank you so much

Regarding the labels which are the MOS, you used the same values for training and testing in your example.

Not sure, about the test values of MOS for further sample training!

**Comment by lyp22** [comment]
In the code, it is just an example showing the file format.
If training, the whole dataset should be downloaded. And the label files should be organized as the given format and put into the code.

**Comment by samanemami** [comment]
Thank you for your kind reply.

You split the SJTU-PCQA WPC2.0 in a way that training has 0.75, and the test has the rest of the reference samples.
Here, you split only the references, not the distorted images.
Hence, have you only trained the model with the reference?

Thank you

**Comment by lyp22** [comment]
It means that 75% of the reference samples and their distorted ones belong to the training set. And the remaining reference samples and their distorted ones belong to the testing set.

**Comment by samanemami** [comment]
Yes, it seems fine, but the MOS is only available for the References point clouds with 40 instances!
